starlette/testclient.py

```diff
@@ -468,19 +474,19 @@ def _choose_redirect_arg(
     def request(  # type: ignore[override]
         self,
         method: str,
-        url: httpx._types.URLTypes,
+        url: URL,
         *,
-        content: httpx._types.RequestContent | None = None,
-        data: _RequestData | None = None,
-        files: httpx._types.RequestFiles | None = None,
+        content: typing.Any = None,
+        data: typing.Any = None,
+        files: typing.Any = None,
         json: typing.Any = None,
-        params: httpx._types.QueryParamTypes | None = None,
-        headers: httpx._types.HeaderTypes | None = None,
-        cookies: httpx._types.CookieTypes | None = None,
-        auth: httpx._types.AuthTypes | httpx._client.UseClientDefault = httpx._client.USE_CLIENT_DEFAULT,
+        params: QueryParams | None = None,
+        headers: Headers | None = None,
+        cookies: Cookies | None = None,
+        auth: Auth | httpx._client.UseClientDefault = httpx._client.USE_CLIENT_DEFAULT,
```

Comment:
Possibly...

```python
UseClientDefault = type(httpx.USE_CLIENT_DEFAULT)

...

auth: Auth | UseClientDefault = USE_CLIENT_DEFAULT
```

(I don't much like this style and I'd enjoy cleaning it up, tho would that be sufficient for now?)

Reply:
I don't think it works for the type checkers.. Does it?

Reply:
That doesn't work.